templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -185,6 +186,11 @@ contents:
         bond_opts=$(nmcli --get-values bond.options conn show ${old_conn})
         if [ -n "$bond_opts" ]; then
           extra_phys_args+=( bond.options "${bond_opts}" )
+          MODE_REGEX=",[[:blank:]]*mode[[:blank:]]*=[[:blank:]]*active-backup[[:blank:]]*,"
+          MAC_REGEX=",[[:blank:]]*fail_over_mac[[:blank:]]*=[[:blank:]]*(1|active)[[:blank:]]*,"
+          if [[ $bond_opts =~ $MODE_REGEX ]] && [[ $bond_opts =~ $MAC_REGEX ]]; then
```

Comment:
Regex issues.

We might not have a leading `,`

`bond_opts=mode=active-backup,fail_over_mac=1,miimon=100`

doesn't match due to no leading `,` before `mode=`

And I guess we might not have a trailing `,` either.

Could be 

`bond_opts=mode=active-backup,miimon=100,fail_over_mac=1`


```
configure-ovs.sh[1953]: + extra_phys_args=()
configure-ovs.sh[2080]: ++ nmcli --get-values connection.type conn show a661c652-5ada-3efd-9deb-f73f9d08a896
configure-ovs.sh[1953]: + '[' bond == vlan ']'
configure-ovs.sh[2084]: ++ nmcli --get-values connection.type conn show a661c652-5ada-3efd-9deb-f73f9d08a896
configure-ovs.sh[1953]: + '[' bond == bond ']'
configure-ovs.sh[1953]: + iface_type=bond
configure-ovs.sh[2088]: ++ nmcli --get-values bond.options conn show a661c652-5ada-3efd-9deb-f73f9d08a896
configure-ovs.sh[1953]: + bond_opts=mode=active-backup,fail_over_mac=1,miimon=100
configure-ovs.sh[1953]: + '[' -n mode=active-backup,fail_over_mac=1,miimon=100 ']'
configure-ovs.sh[1953]: + extra_phys_args+=(bond.options "${bond_opts}")
configure-ovs.sh[1953]: + MODE_REGEX=',[[:blank:]]*mode[[:blank:]]*=[[:blank:]]*active-backup[[:blank:]]*,'
configure-ovs.sh[1953]: + MAC_REGEX=',[[:blank:]]*fail_over_mac[[:blank:]]*=[[:blank:]]*(1|active)[[:blank:]]*,'
configure-ovs.sh[1953]: + [[ mode=active-backup,fail_over_mac=1,miimon=100 =~ ,[[:blank:]]*mode[[:blank:]]*=[[:blank:]]*active-backup[[:blank:]]*, ]]
configure-ovs.sh[1953]: + '[' '!' '' = 0 ']'
configure-ovs.sh[1953]: + extra_phys_args+=(802-3-ethernet.cloned-mac-address "${iface_mac}")
configure-ovs.sh[1953]: + nmcli connection show ovs-if-phys0
configure-ovs.sh[1953]: + ovs-vsctl --timeout=30 --if-exists destroy interface bond0
```

Reply:
I removed the blanks in the regex as well as nmcli gives us normalized output.

Reply:
I guess it might look like I haven't tested but I did, I guess that the commas were a last minute addition :facepalm:
I tested again with a couple of bonds modes with and without mac address cloning.
Tested a couple of teaming modes as well.